### ISSUE TITLE
Any Explanations about conv_tbc ? Thanks ~~

### ISSUE THREAD

**Issue by mali-nuist** [pytorch/fairseq#172]
I read the source code but i can't figure out what the operation of conv_tbc  is doing , Any explanations ?
For example:
  If the  shape of input tensor  is [10,20](batch_size is 10 and max sentence length is 20) , and then the word embedding resulted in the embed tensor size of [10,20,256](embed size is 256). Then the transposed tensor of size [20,10,256] will be fed into the cov as: x = conv(x)(the conv input channel is 256 and output channel is 512) , which resulted in a new tensor of shape[20, 10, 512] . What the conv is doing ? it seems that the conv treated the embedding axis as channel ?

**Comment by myleott** [comment]
conv_tbc is the same as torch.nn.Conv1d, but accepts a different input shape.

The input shape for nn.Conv1d is batch x channels x time (BCT), which would require a transpose since the rest of the network operates with time x batch x channel (TBC) tensors. conv_tbc takes time x batch x channel (TBC) input directly.

**Comment by mali-nuist** [comment]
thanks for your reply!  i will check the documentation about the torch.nn.Conv1d!

**Comment by mali-nuist** [comment]
thanks for you reply ! i will check the documentation about the torch.nn.Conv1d
